src/transaction/index.ts

```diff
@@ -426,7 +392,10 @@ export async function transaction<
       t.delete(firebaseDoc)
     }
 
-    return readFunction({ get }).then((data) =>
+    return readFunction({
+      get: (...props) => (getCommon as any)(props[0], props[1], props[2], {a, t}),
```

Comment:
I had to `as any` because all those generics and overloads of get weren't matching, but it's the same params order. Tried to get it working without the type cast but hadn't success.

Also, I `props[0], props[1], props[2]` instead of `...props` because some of them are optional and would make `{a, t}` be in the wrong param instead of the forth one.